rts/Rendering/GL/TexBind.h

```diff
@@ -0,0 +1,38 @@
+/* This file is part of the Recoil engine (GPL v2 or later), see LICENSE.html */
+
+#pragma once
+
+#include "myGL.h"
+#include "Rendering/GlobalRendering.h"
+
+namespace GL
+{
+
+class TexBind {
+public:
+	inline TexBind(int slot, GLenum target, GLuint textureID)
+	:	texUnit(GL_TEXTURE0 +GLenum(slot >= 0? slot : globalRendering->maxCombShSlots+slot)),
```

Comment:
Can you explain `globalRendering->maxCombShSlots+slot` part?

Reply:
Sure. MaxCombShSlots is a number of texture units for all shader stages. It is usually a few times the number of texture units per shader stage, so much more than 16. gl4 guarantees this value to be at least 80.
I use this value as a maximum index for glActiveTexture.
Specs ref:
https://registry.khronos.org/OpenGL-Refpages/gl4/html/glActiveTexture.xhtml
(GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS)

Reply:
We discussed that elsewhere and I don't like that approach: slot should either be explicitly mentioned or default to the first 16 slots.
I can see a few issues with `globalRendering->maxCombShSlots+slot` approach:
1) A management of uniform values (if needed) corresponding to such texture becomes cumbersome
2) Knowing how the spec is executed in fact by some lesser drivers, I don't want to try our luck.

Reply:
> We discussed that elsewhere and I don't like that approach: slot should either be explicitly mentioned or default to the first 16 slots. I can see a few issues with `globalRendering->maxCombShSlots+slot` approach:
> 
> 1. A management of uniform values (if needed) corresponding to such texture becomes cumbersome
> 2. Knowing how the spec is executed in fact by some lesser drivers, I don't want to try our luck.

By driver issues, do you mean that glGet(GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS) may return a value higher than it actually _can_ use? So glActiveTexture(GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS-1) will produce error. In that case, it's a problem.
Are there drivers that actually do that?

Reply:
Using just fixed 16 or 32 slot has another problems, that's why I am trying to figure it out.

Reply:
I am having a problem understanding how a driver possibly failing at that works at all.

Reply:
If such drivers are confirmed to exist, I have no problem reverting to the old idea, where we just re-bind previous texture, as it is a cleaner solution than fixed slot.

Reply:
It isn't worth time investment to figure out
Reverted to old method. It should be safe to use